### PR TITLE
Optimized scan & Configurable file patterns

### PR DESCRIPTION
**Pull request by ktamas77**
- Dynamic swagger generation is now pre-scans files for OpenApi definitions before it's passing it to the Tokenizer in order to speed up dynamic API generation for large repositories
- Configurable Patterns added

### CODE REVIEW

File: src/functions.php [hunk 1]
```diff
@@ -17,12 +17,24 @@
...
 
+    /**
+     * Determines whether the given string has openapi definitions or not
+     *
+     * @param string $filecontent File content to analyze
+     * @return boolean
+     */
+    function hasOpenApiDefinitions($filecontent)
+    {
+        return strpos($filecontent, '@OA\\') !== false;
```
Comment (penja) [inline src/functions.php]: This will not work for the custom annotations under other namespaces

Comment (ktamas77) [reply src/functions.php]: true - i'll close this PR

Comment (penja) [reply src/functions.php]: You can try get info from `Analyser::defaultImports` and `Analyser:whitelist` properties. But we should investigate this and coverage all cases with tests.

**Comment by bfanger** [comment]
Some features like inheritance rely on parsing intermediate files which might not have annotations. 
But If your project doen't, using **grep** with **xargs** you can apply the same optimisation:

```sh
grep --null -r -l --include "*.php"  "@OA" $my-project-directory | xargs -0 openapi
```
I consider swagger-php fast enough, and therefor closing this PR, but thanks for the ideas.

**Pull request closed without merging**